lib/data/tools/select.dart

```diff
@@ -30,6 +30,33 @@ class Select extends Tool {
     selectResult.pageIndex = -1;
   }
 
+  Color? getDominantStrokeColor(){
+    Map<Color, int> colorDistribution = <Color, int>{};
+    for(Stroke stroke in selectResult.strokes){
+      int strokeSize = stroke.polygon.length;
+
+      colorDistribution.update(stroke.strokeProperties.color, (value) => value+strokeSize, ifAbsent: () {
+        return strokeSize;
+      },);
+    }
+
+    if(colorDistribution.isEmpty){
+      return null;
+    }
+
+    Color dominantColor = colorDistribution.keys.first;
+    int maxSize = colorDistribution.values.first;
+
+    colorDistribution.forEach((color, size) {
+      if(maxSize < size){
+        maxSize = size;
+        dominantColor = color;
+      }
+    });
```

Comment:
This could be cleaner as a `reduce` call if `Map` supports it (perhaps with `map.entries`).